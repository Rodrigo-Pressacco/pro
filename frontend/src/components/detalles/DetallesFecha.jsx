import React, { useContext,useEffect,useState } from "react";
import { Link} from "react-router-dom";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import "../../styles/detalles/DetallesFechas.css";
import actualizarFechas from "../extras/ActualizarFechas";
import fechasContext from "../../contexts/fechasContext";

const DetallesFecha = (props) => {
    const hoy = new Date()
    let maximaFecha = new Date(hoy)
    maximaFecha.setDate(hoy.getDate() + 365) // Al igual que con vuelos, lo maximo de anticipado que se puede elegir es un anio
    
    const [dateRange, setDateRange] = useState([null, null]);
    const [startDate, endDate] = dateRange;
    const [minDate, setMinDate] = useState(hoy)
    const [maxDate, setMaxDate] = useState(maximaFecha)
    const context = useContext(fechasContext)
    
    return (
        <div className="div-general">
            <h3>Fechas Disponibles</h3>
            <div className="div-calendario-boton">
                <div className="div-detalles-calendario">
                    <DatePicker
                        selectsRange={true}
                        startDate={startDate}
                        endDate={endDate}
                        monthsShown={2}
                        dateFormat=" dd 'de' MMM. "
                        className="form-calendarDate"
                        onChange={(update) => {
                            setDateRange(update)
                            if (update[1] === null) {
                                setMinDate(update[0]);
                                setMaxDate(actualizarFechas(props.fechasOcupadas, update[0]));
                                context.setFechaInicial(hoy);
                                context.setFechaFinal(hoy);
                            } else {
                                setMinDate(hoy)
                                setMaxDate(maximaFecha)
                                context.setFechaInicial(update[0]);
                                context.setFechaFinal(update[1]);
                            }
                            // Al seleccionar una fecha, se eliminan todas las fechas anteriores, cuando completo el rango, se vuelve a la normalidad
                            // update[1] === null ? setMinDate(update[0]) : setMinDate(hoy)
                            // Al Seleccionar una fecha, se eliminan todas las fechas posteriores a la primer reserva despues de la fecha, cuando completo el rango, se vuelve a la normalidad
                            // update[1] === null ? setMaxDate(actualizarFechas(props.fechasOcupadas, update[0])) : setMaxDate(maximaFecha)
                            }
                        }
                        minDate={minDate}
                        maxDate={maxDate}
                        excludeDates={props.fechasOcupadas}
                        inline

                    />
                </div>
                <div className="div-detallesFecha-button">
                    <p>Agrega tus fechas de viaje para obtener precios exactos</p>
                    {props.logueado?<Link to={`/detalle/${props.id}/reserva`} datos={props.id} >Iniciar reserva</Link>:
                    <Link to={`/login`} >Iniciar reserva</Link>}
                    
                </div>
            </div>
        </div>
    )
}

export default DetallesFecha;
