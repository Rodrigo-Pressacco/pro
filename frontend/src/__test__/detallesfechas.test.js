import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import DetallesFecha from '../components/detalles/DetallesFecha'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<DetallesFecha />', () => {
    let fechas

    beforeEach(() => {
        fechas = shallow(<DetallesFecha />)
    });

    it("Deberia mostrar <DetallesFecha/> correctamente", () => {
        expect(fechas).toMatchSnapshot();
    });

    it("Deberia retornar 1 etiqueta h3", () => {
        let node = fechas.find("h3")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 1 etiqueta p", () => {
        let node = fechas.find("p")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 1 etiqueta Link", () => {
        let node = fechas.find("Link")
        expect(node.length).toEqual(1)
    });

})