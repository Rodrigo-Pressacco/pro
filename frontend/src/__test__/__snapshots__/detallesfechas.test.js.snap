// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetallesFecha /> Deberia mostrar <DetallesFecha/> correctamente 1`] = `
<div
  className="div-general"
>
  <h3>
    Fechas Disponibles
  </h3>
  <div
    className="div-calendario-boton"
  >
    <div
      className="div-detalles-calendario"
    >
      <r
        allowSameDay={false}
        className="form-calendarDate"
        customTimeInput={null}
        dateFormat=" dd 'de' MMM. "
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        enableTabLoop={true}
        endDate={null}
        excludeScrollbar={true}
        focusSelectedMonth={false}
        inline={true}
        maxDate={2022-11-24T22:39:35.730Z}
        minDate={2021-11-24T22:39:35.730Z}
        monthsShown={2}
        nextMonthAriaLabel="Next Month"
        nextMonthButtonLabel="Next Month"
        nextYearAriaLabel="Next Year"
        nextYearButtonLabel="Next Year"
        onBlur={[Function]}
        onCalendarClose={[Function]}
        onCalendarOpen={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthAriaLabel="Previous Month"
        previousMonthButtonLabel="Previous Month"
        previousYearAriaLabel="Previous Year"
        previousYearButtonLabel="Previous Year"
        readOnly={false}
        renderDayContents={[Function]}
        selectsRange={true}
        shouldCloseOnSelect={true}
        showFourColumnMonthYearPicker={false}
        showFullMonthYearPicker={false}
        showMonthYearPicker={false}
        showPopperArrow={true}
        showPreviousMonths={false}
        showQuarterYearPicker={false}
        showTimeInput={false}
        showTimeSelect={false}
        showTwoColumnMonthYearPicker={false}
        showYearPicker={false}
        startDate={null}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
        yearItemNumber={12}
      />
    </div>
    <div
      className="div-detallesFecha-button"
    >
      <p>
        Agrega tus fechas de viaje para obtener precios exactos
      </p>
      <Link
        to="/login"
      >
        Iniciar reserva
      </Link>
    </div>
  </div>
</div>
`;
